docs/README.md

```diff
@@ -53,6 +60,25 @@ To make changes to the docs, simply make a Pull Request with your
 proposed changes as normal. When the PR is merged the docs will be
 automatically updated.
 
+## Including Source Code
+
+We want to make sure that all source code in the documentation is tested as part of the build and release process. We
```

Comment:
I wonder if you considered the opposite:

Leaving the code example inlined in the documentation, and extracting it out to `datafusion-docs-test` with a script. 

This would have the benefit that the markdown would be closer to the actual output documentation, and it might be easier to work on the examples inline.

The downside is that then we would need to somehow run a script to copy the examples into datafusion-docs-tests as part of CI to make sure they were up to date

Reply:
This approach is more challenging because sometimes we are copying a whole function, and sometimes just a few lines of code, and some snippets depend on other snippets. It is also hard to write and test code with this approach because there is no IDE support.